src/pip/_vendor/resolvelib/resolvers.py

```diff
@@ -368,11 +368,12 @@ def resolve(self, requirements, max_rounds):
             if failure_causes:
                 # Backtrack if pinning fails. The backtrack process puts us in
                 # an unpinned state, so we can work on it in the next round.
+                causes = [i for c in failure_causes for i in c.information]
+                self._r.start_backtracking(causes)
```

Comment:
obviously waiting for resolvelib PR & release, just adding this to see that I'm on the right path.
https://github.com/sarugaku/resolvelib/pull/81